uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/controller/request/UniProtKBIdsDownloadRequest.java

```diff
@@ -0,0 +1,31 @@
+package org.uniprot.api.uniprotkb.controller.request;
+
+import lombok.Data;
+
+import org.uniprot.api.rest.request.IdsSearchRequest;
+import org.uniprot.api.rest.validation.ValidDownloadByIdsRequest;
+import org.uniprot.store.config.UniProtDataType;
+
+@Data
+@ValidDownloadByIdsRequest(uniProtDataType = UniProtDataType.UNIPROTKB)
+public class UniProtKBIdsDownloadRequest implements IdsSearchRequest {
+    private String accessions;
+    private String fields;
+    private String download;
+    private Integer size;
+    private String cursor;
+
+    public String getCommaSeparatedIds() {
+        return this.accessions;
+    }
+
+    @Override
+    public String getFacetFilter() {
```

Comment:
I think we need to accept facetFilter, when download= false, right?
If you agree, we can not return always null

Reply:
See my reply above.